### PR TITLE
Ak/new oauth flow

### PR DESCRIPTION
**Pull request by mitochondrion**
@jsom

### CODE REVIEW

**Comment by jsom** [comment]
@mitochondrion lol

File: ExampleApp/AppDelegate.swift [hunk 1]
```diff
@@ -1,21 +1,65 @@
...
-
+    static let API_KEY = "tradeit-fx-test-api-key" //"tradeit-test-api-key"
+    static let ENVIRONMENT = TradeItEmsTestEnv
     var window: UIWindow?
+    let tradeItLauncher: TradeItLauncher
 
-    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
-        // Override point for customization after application launch.
+    override init() {
+        tradeItLauncher = TradeItLauncher(apiKey: AppDelegate.API_KEY, environment: AppDelegate.ENVIRONMENT)
```
Comment (jsom) [inline ExampleApp/AppDelegate.swift]: `TradeItSDK.launcher`?

File: ExampleApp/AppDelegate.swift [hunk 2]
```diff
@@ -1,21 +1,65 @@
 import UIKit
+@testable import TradeItIosTicketSDK2
```
Comment (jsom) [inline ExampleApp/AppDelegate.swift]: `@testable`?

File: ExampleApp/Info.plist [hunk 3]
```diff
@@ -2,6 +2,17 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>CFBundleURLTypes</key>
+	<array>
+		<dict>
+			<key>CFBundleURLName</key>
+			<string>omgwtfbbq</string>
```
Comment (jsom) [inline ExampleApp/Info.plist]: Use something TradeIt?

File: TradeItIosEmsApi/TradeItConnector.h [hunk 4]
```diff
@@ -116,4 +116,16 @@
...
+                                 completionBlock:(void (^ _Nullable)(TradeItResult * _Nullable))completionBlock;
+
+- (void)getOAuthLoginPopupURLForTokenUpdateWithBroker:(NSString * _Nullable)broker
+                                               userId:(NSString * _Nullable)userId
+                              interAppAddressCallback:(NSString * _Nullable)interAppAddressCallback
+                                      completionBlock:(void (^ _Nullable)(TradeItResult * _Nullable))completionBlock;
+
+- (void)getOAuthAccessTokenWithOAuthVerifier:(NSString * _Nullable)oAuthVerifier
+                             completionBlock:(void (^ _Nullable)(TradeItResult * _Nullable))completionBlock;
+
```
Comment (jsom) [inline TradeItIosEmsApi/TradeItConnector.h]: Could you have done a Swift extension here instead?

Comment (mitochondrion) [inline TradeItIosEmsApi/TradeItConnector.h]: Uh, probably, but I wasn't adding anything Swift specific.

Comment (jsom) [inline TradeItIosEmsApi/TradeItConnector.h]: Sure, if you want to maintain more Obj-C code? There are a bunch of new classes in this PR that could have been Swift

File: TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift [hunk 5]
```diff
@@ -23,6 +24,122 @@ import PromiseKit
...
+    public func getOAuthLoginPopupUrl(withBroker broker: String,
+                                      deepLinkCallback: String,
+                                      onSuccess: @escaping (_ oAuthLoginPopupUrl: String) -> Void,
+                                      onFailure: @escaping (TradeItErrorResult) -> Void) {
+        self.connector.getOAuthLoginPopupUrlForMobile(withBroker: broker,
+                                                      interAppAddressCallback: deepLinkCallback) { tradeItResult in
+            switch tradeItResult {
+            case let oAuthLoginPopupUrlForMobileResult as TradeItOAuthLoginPopupUrlForMobileResult:
+                self.currentOAuthBroker = broker
+                onSuccess(oAuthLoginPopupUrlForMobileResult.oAuthURL ?? "")
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift]: Returning on success and empty string if `oAuthURL` is missing??

**Pull request closed without merging**